### PR TITLE
chore: update Table styles

### PR DESCRIPTION
**Pull request by paulocfjunior**
#### Description

Closes https://github.com/traefik/faency/issues/152.

Updates the Table and related components' style.

Light:
Screen Shot 2022-02-02 at 18 27 57

Dark:
Screen Shot 2022-02-02 at 18 27 50

### CODE REVIEW

**Review by seedy — APPROVED** [review]
LGTM

**Review by gndz07 — APPROVED** [review]

**Comment by seedy** [comment]
:information_source: I'm taking care of the rebase

File: components/Table/Table.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { styled, VariantProps } from '../../stitches.config';
+import { elevationVariants } from '../Elevation';
 import { Label } from '../Label';
 
 export const Caption = styled('caption', {
```
Comment (seedy) [inline components/Table/Table.tsx:5]: @paulocfjunior We didn't update table `Caption` styles. To me the `Caption` should the table's title.

Mockups aren't integrating it at all.

I propose that we take our existing usage and apply its styles to the `Caption`. We could inherit from `Heading` for example.

WDYT?

**Review by seedy — CHANGES_REQUESTED** [review]
One change to apply coming from rebase. Don't hesitate to ping me if you wish to work with me on this.

File: components/Table/Table.tsx (hunk 1 above)
Comment (paulocfjunior) [reply components/Table/Table.tsx]: Yes, sounds good to me 👌

**Review by seedy — APPROVED** [review]

**Pull request merged**